CLI 1.0.0 release notes + doc changes (#108)
astronomer/cloud-docs: docs/cli-release-notes.md

If you have any questions or a bug to report, don't hesitate to reach out to us via Slack or Intercom. We're here to help.

## v1.0.0

Release date: September 28, 2021

> - Improvement: `astro dev init` now always pulls in the latest version of Astronomer Runtime for newly scaffolded Astronomer projects. This means that you no longer have to upgrade the CLI in order to take advantage of a new Runtime release. Note that you still need to manually [upgrade Runtime](upgrade-runtime) for existing projects.

Reviewer: ```suggestion
- Improvement: For new projects, `astro dev init` now always pulls the latest version of Astronomer Runtime. This means that new users are no longer dependent on a specific version of the CLI in order to take advantage of a new Runtime release. For existing projects, you still need to manually [upgrade Runtime](upgrade-runtime).
```

Author: Not sure how the changes here are helping with the meaning of the release note, and not sure that the benefit of this change applies only to new users. Going to stick with the original wording for now.
